routers/web/repo/setting/setting.go

```diff
@@ -476,9 +477,10 @@ func SettingsPost(ctx *context.Context) {
 			deleteUnitTypes = append(deleteUnitTypes, unit_model.TypeWiki)
 		} else if form.EnableWiki && !form.EnableExternalWiki && !unit_model.TypeWiki.UnitGlobalDisabled() {
 			units = append(units, repo_model.RepoUnit{
-				RepoID: repo.ID,
-				Type:   unit_model.TypeWiki,
-				Config: new(repo_model.UnitConfig),
+				RepoID:             repo.ID,
+				Type:               unit_model.TypeWiki,
+				Config:             new(repo_model.UnitConfig),
+				EveryoneAccessMode: perm.ParseAccessMode(form.DefaultWikiEveryoneAccess, perm.AccessModeUnset, perm.AccessModeRead, perm.AccessModeWrite),
```

Comment:
All other units will be hidden when updating. I think for other unit, the default value becomes 0 which should be -1?

Reply:
<img width="928" alt="image" src="https://github.com/go-gitea/gitea/assets/81045/e30d05fc-3041-42b6-8cc6-e518803130e6">
All other units disappeared.

Reply:
> All other units will be hidden when updating. I think for other unit, the default value becomes 0 which should be -1?

How to reproduce? Actually -1 or 0 doesn't have difference, they are all ignored.

Reply:
Hmm, I see the problem, maybe it is caused by the UnitsMode, checking it.

Reply:
Just create a new repository and then go to setting, chose read and save. Then you can see the above problem.

Reply:
The problem is that the units are always loaded now. So the perm.AccessMode is ignored. Will make a fix.

Reply:
Permission fix is done in 34650ad1b23e287e447ac0deaf3d93fa8f31357e , and the tests are also updated accordingly. Now it is more clear to me that how the "access mode" works.